packages/importer/src/providers/standardnotes/index.ts

```diff
@@ -0,0 +1,228 @@
+        if (language === "htmlmixed") language = "html";
+        let code = hljs.highlightAuto(data, [language]);
+        let html = `<pre class="hljs language-${language}">
+          ${code.value}
+        </pre>`;
+        return {
+          type: ContentType.HTML,
+          data: html,
+        };
+      case "json":
+        if (editorType.jsonFormat === "token") {
+          let tokens = <TokenVaultItem[]>JSON.parse(data);
+
+          let html = `
+          ${tokens.map((token) =>
+            this.buildTableWithRows(
+              Object.keys(token)
+                .map(
+                  (key) => `<tr>
+              <th style="width:130px" >${key}</th>
+              <td>${token[key] || ""}</td>
+            </tr>`
+                )
+                .join("")
+            )
+          )}`;
+          return {
+            data: html,
+            type: ContentType.HTML,
+          };
+        } else {
+          let spreadsheet = <SpreadSheet>JSON.parse(data);
+          let html = ``;
+          for (let sheet of spreadsheet.sheets) {
+            if (sheet.rows.length > 0) {
+              let maxCols =
+                this.maxIndexItem(
+                  sheet.rows.map((row) => {
+                    return this.maxIndexItem(row.cells);
+                  })
+                ).index + 1;
+
+              let maxRows = this.maxIndexItem(sheet.rows).index + 1;
+              let rows = [];
+
+              for (let i = 0; i < maxRows; i++) {
+                let rowAtIndex = sheet.rows.find((row) => row.index === i);
+                // create an empty row to fill index
+                if (!rowAtIndex) rowAtIndex = { index: i, cells: [] };
+                let cells = [];
+                for (let col = 0; col < maxCols; col++) {
+                  let cellAtCol = rowAtIndex.cells.find(
+                    (cell) => cell.index === col
+                  );
+                  // create an empty cell to fill index
+                  if (!cellAtCol) cellAtCol = { value: "", index: col };
+                  cells.push(`<td>${cellAtCol.value || ""}</td>`);
+                }
+                rows.push(`<tr>
+                ${cells.join("")}
+                </tr>`);
+              }
+
+              let table = this.buildTableWithRows(rows.join(""));
+              html = html + table;
+            }
+          }
+          return {
+            type: ContentType.HTML,
+            data: html,
+          };
+        }
+      default:
+        return {
+          data: converter.makeHtml(data),
+          type: ContentType.HTML,
+        };
+    }
+  }
+
+  async process(
+    files: File[],
+    settings: ProviderSettings
+  ): Promise<ProviderResult> {
+    return iterate(this, files, (file, notes) => {
+      if (file.name !== "Standard Notes Backup and Import File.txt")
+        return Promise.resolve(true);
+      let data: BackupType = <BackupType>JSON.parse(file.text);
```

Comment:
Check if the parsed data is valid and includes the required properties.

We should also do a version check here and give error.